### DISCUSSION TITLE
Temperature Chart

### DISCUSSION THREAD

**Discussion by riesi77** [teslamate-org/teslamate#886] [Ideas]
**What would you like to be added**:
Add a new dashboard which shows a chronological chart of the outside temperature. Nice to have: also display the inside temperature. 

**Why is this needed**:
The car collects the information (inside temperature, outside temperature) - This is informative

**Additional context**:

**Comment by cwanja** [comment]
During what periods? 24/7? If so, it would never sleep (I believe). When it falls into a deep sleep, do not believe those values are reported by the API. 

If you want it during drivers or charges, that is available.

**Comment by DrMichael** [comment]
It is indeed. I've put together a temperature panel:

![2020-09-01 15_46_01-Window](https://user-images.githubusercontent.com/8087682/91859209-71505500-ec6a-11ea-95f8-14ff9835b39b.png)

Import this (after unzipping of course :-):

[Temperature-1598968166979.zip](https://github.com/adriankumpf/teslamate/files/5156597/Temperature-1598968166979.zip)

**Comment by bahree** [comment]
Thoughts on adding this as part of the default install @adriankumpf ?

**Comment by DrMichael** [comment]
Well, it is a stripped down version of my statistics panel:

![2020-09-01 18_01_49-Statistics - Grafana](https://user-images.githubusercontent.com/8087682/91876656-3f94b980-ec7d-11ea-9e45-8531db864af3.png)

That still has the issue, that you have to change the postgresql server tiimezone to your timezone... See https://github.com/adriankumpf/teslamate/issues/664 (and would have to be migrated to the new table in Grafana 7, waiting for the bug fixes).

Happy to share anyway.

[Statistics-1598976191847.zip](https://github.com/adriankumpf/teslamate/files/5157493/Statistics-1598976191847.zip)

**Comment by adriankumpf** [comment]
> Thoughts on adding this as part of the default install @adriankumpf ?

I am not opposed to adding it, but I do not understand its purpose. Since the car sleeps most of the time and is therefore not available through the API, the dashboard only shows the average temperature during trips. What do you do with this information?

The statistics panel looks pretty cool though. A PR would be appreciated @DrMichael but I don't know how to work around the timezone issue.

**Comment by cwanja** [comment]
> Well, it is a stripped down version of my statistics panel:
> 
> ![2020-09-01 18_01_49-Statistics - Grafana](https://user-images.githubusercontent.com/8087682/91876656-3f94b980-ec7d-11ea-9e45-8531db864af3.png)
> 
> That still has the issue, that you have to change the postgresql server tiimezone to your timezone... See #664 (and would have to be migrated to the new table in Grafana 7, waiting for the bug fixes).
> 
> Happy to share anyway.
> 
> [Statistics-1598976191847.zip](https://github.com/adriankumpf/teslamate/files/5157493/Statistics-1598976191847.zip)

This is definitely really interesting statistics. Hopefully there is a way to set the timezone either through the interface or natively based upon the host.

**Comment by DrMichael** [comment]
> This is definitely really interesting statistics. Hopefully there is a way to set the timezone either through the interface or natively based upon the host.

Well, to make things worse, you can set the display timezone in Grafana 7 to every timezone instead of just UTC and local timezone...

**Comment by DrMichael** [comment]
> The statistics panel looks pretty cool though. A PR would be appreciated @DrMichael but I don't know how to work around the timezone issue.

I would hold back, until it is running properly with Grafana 7... Still two bugs with hidden fields and formatting timestamps. And still have to figure out, how to handel the km/mi and c/f settings reasonably, @adriankumpf any idea there?

**Comment by adriankumpf** [comment]
Yes, it's a real pity that the new component still doesn't support hidden fields. Can the timestamp problem solved with a custom format (https://github.com/grafana/grafana/issues/24917)?

What's the issue with units again? The existing conversion helpers do not work?

**Comment by DrMichael** [comment]
> Can the timestamp problem solved with a custom format ([grafana/grafana#24917](https://github.com/grafana/grafana/issues/24917))?

Well, it turns out, that formatting a time string does not work. It does work though for timestamps with "time: YYY-MM-DD HH:mm" e.g.

> What's the issue with units again? The existing conversion helpers do not work?

Ah, you can deal with that by using regexes... But, it appears to me, that the "new" fields (e.g. when you switch km to mi) show up at the far right of the table.

What are the conversion helpers?

**Comment by adriankumpf** [comment]
> Ah, you can deal with that by using regexes... But, it appears to me, that the "new" fields (e.g. when you switch km to mi) show up at the far right of the table.
> 
> What are the conversion helpers?

Ah ok, I thought you meant the conversion from km to mi based on user settings.

**Comment by DrMichael** [comment]
Oh, yes, I use e.g. "convert_km(sum(distance)::numeric, '$length_unit')" in queries...

(just to make others aware of that... :-)

**Comment by akac** [comment]
What id like to see is similar to this chart in Stats app:
https://pbs.twimg.com/media/EmVc8JRVcAE6uEz?format=jpg&name=large

Essentially it’s a temperature/efficiency chart.

**Comment by Cyberwizzard** [comment]
This sounds very useful - if only when the car is awake.

Efficiency chart is cool but also as a simple plot over time (like the charge in the battery); to illustrate how warm/cold it was when the car was used.

A bit more esoteric would be a temperature per location over a trip - that would be a stretch for simple usage but I would find it interesting to see the temperature gradient while driving down to the city for example.

**Comment by Kabouik** [comment]
A bit late to the party, but I think discarding a temperature logging dashboard due to the car sleeping would be a mistake. Sentry mode keeps the car online all the time, and large amounts of data can be collected this way. Those whose car sleeps often can just not use this dashboard much, and temperature while driving only would still be interesting. 

Speaking of, would there be a way to add a hour period to @DrMichael's dashboard? The car would be a great way for me to get geolocalized hourly data.

**Comment by Kabouik** [comment]
With the shameful assistance of AI, I could get something that starts looking interesting:

![Screenshot_20251228-005221~2](https://github.com/user-attachments/assets/b8ba49e8-dc9d-4757-aaa9-3b7a62257471)
![Screenshot_20251228-005302~2](https://github.com/user-attachments/assets/46d51d20-9111-4492-98ca-f163c755ce56)
![Screenshot_20251228-005335~2](https://github.com/user-attachments/assets/d399edbf-e80d-45b1-932b-145d58557730)

We can clearly see the drive when inside temperature increased, while the rest was at parking (with sentry mode on, so no sleep). 

I'm not posting the json because it is not fully functional yet, especially the map which is pretty barebones, but the dashboard allows showing raw samples or aggregating to buckets and seems to work fine at that. What would be nice would be to track on the map as the cursor moves in the top plot or table, and use the map to restrict the sample area to plot. 

That's an expensive thermometer, but could be useful in some scenarios.

**Comment by Kabouik** [comment]
Here is an attempt ready to test:

All samples:
<img width="1850" height="818" alt="ss20251228-023438b" src="https://github.com/user-attachments/assets/d550da45-45d8-4e07-9d0a-5ebe6fe60c4d" />

6-hour averages:
<img width="1844" height="809" alt="ss20251228-023502b" src="https://github.com/user-attachments/assets/07e03cb0-1aed-488b-ad80-5f229a645b01" />

[Temperature logs-1767144817914.json](https://github.com/user-attachments/files/24389261/Temperature.logs-1767144817914.json)

I hope it's not too broken, but don't expect much, it's AI-generated and not proof-read. ~~The main limitation I noticed so far is the map cannot be synchronized with the plot cursor when binning data in buckets, hence the button to toggle to all raw samples. Not ideal.~~ 

Edits:
- Fixed binning issue with map
- Data shown in decending time order in the table
- Added support for car states in plot and table:
  ![Screenshot_20251228-173131~2](https://github.com/user-attachments/assets/fdf0bc67-5ba3-4b53-9613-315e5165ab4a) ![Screenshot_20251228-173120~2](https://github.com/user-attachments/assets/b93afaa5-01d7-4ab4-9891-66418e4dfc3d)
